pkg/minishift/openshift/openshift.go

```diff
@@ -225,3 +252,53 @@ func deleteBackup(target OpenShiftPatchTarget, id string, commander docker.Docke
 
 	return nil
 }
+
+func getLocalConfigFile(target string) string {
+	openshiftVersion := getOpenshiftVersion()
+	switch target {
+	case "master":
+		valid, _ := openshiftVersionCheck.IsGreaterOrEqualToBaseVersion(openshiftVersion, constants.RefactoredOcVersion)
+		if !valid {
+			return "/var/lib/minishift/openshift.local.config/master/master-config.yaml"
+		}
+		return "/var/lib/minishift/base/openshift-apiserver/master-config.yaml"
+	case "node":
+		valid, _ := openshiftVersionCheck.IsGreaterOrEqualToBaseVersion(openshiftVersion, constants.RefactoredOcVersion)
+		if !valid {
+			return "/var/lib/minishift/openshift.local.config/node-localhost/node-config.yaml"
+		}
+		return "/var/lib/minishift/base/node/node-config.yaml"
+	}
+	return ""
+}
+
+func getContainerConfigFile(target string) string {
+	openshiftVersion := getOpenshiftVersion()
+	switch target {
+	case "master":
+		valid, _ := openshiftVersionCheck.IsGreaterOrEqualToBaseVersion(openshiftVersion, constants.RefactoredOcVersion)
+		if !valid {
+			return "/var/lib/origin/openshift.local.config/master/master-config.yaml"
+		}
+		return "/etc/origin/master/master-config.yaml"
+	case "node":
+		valid, _ := openshiftVersionCheck.IsGreaterOrEqualToBaseVersion(openshiftVersion, constants.RefactoredOcVersion)
+		if !valid {
+			return "/var/lib/origin/openshift.local.config/node-localhost/node-config.yaml"
+		}
+		return "/var/lib/origin/openshift.local.config/node/node-config.yaml"
+	}
+	return ""
+}
+
+func getOpenshiftVersion() string {
+	var openshiftVersion string
+
+	jsonDataPath := filepath.Join(constants.Minipath, "machines", constants.MachineName+".json")
+	instanceState.InstanceConfig, _ = instanceState.NewInstanceConfig(jsonDataPath)
+
+	if instanceState.InstanceConfig != nil {
+		openshiftVersion = instanceState.InstanceConfig.OpenshiftVersion
+	}
+	return openshiftVersion
+}
```

Comment:
Logically 
```
if instanceState.InstanceConfig != nil {
		openshiftVersion = instanceState.InstanceConfig.OpenshiftVersion
	}
``` 
should come first in the function and if is `nil` then create an instancState.